### PR TITLE
➕ 💅 Add some base components 

### LINKED ISSUE #3: Create Components
Frontend Components

- [x] Cards
- [x] Badges
- [x] Header
- [x] Buttons
- [x] Inputs

### PR DESCRIPTION
**Pull request by bwnstck**
# Description
Start to add some components like badges or buttons

Fixes #3 

## Type of change
- [x ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: client/.storybook/preview.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
+import "!style-loader!css-loader!./main.css";
```
Comment (lmachens) [inline client/.storybook/preview.js:1]: I guess you like to import the same styles as in your App?
Instead, you can add a [global decorator ](https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) and import `GlobalStyles`.
You can remove the `main.css` afterwards.

File: client/src/App.js [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
+--carnelian: hsla(0, 75%, 40%, 1);
+--dark-red: hsla(0, 86%, 29%, 1);
+--electric-indigo: hsla(259, 82%, 55%, 1);
+  }
+  #root {
+    margin: 0;
+    padding: 0;
+    font-family: Open-Sans, Helvetica, Sans-Serif;
+  }
+`;
```
Comment (lmachens) [inline client/src/App.js:27]: Format is not applied here

File: client/src/App.js [hunk 3]
```diff
@@ -1,25 +1,40 @@
 import React from "react";
-import logo from "./logo.svg";
-import "./App.css";
+import { createGlobalStyle } from "styled-components/macro";
+
+import { Badge } from "./stories/Badge";
+import { HeaderMain } from "./stories/HeaderMain";
+
```
Comment (lmachens) [inline client/src/App.js:6]: Never import your stories from your app or components.
The stories are only for Storybook.

File: client/src/App.test.js [hunk 4]
```diff
@@ -1,9 +1,7 @@
...
+function sum(a, b) {
+  return a + b;
+}
 
-test("renders learn react link", () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+test("adds 1 + 2 to equal 3", () => {
+  expect(sum(1, 2)).toBe(3);
```
Comment (lmachens) [inline client/src/App.test.js:6]: This test doesn't make much sense here, especially if the file is named `App.test.js`.
You can just test if the App renders:

```js
test("renders App", () => {
  render(<App />);
});
```

File: client/src/stories/Badge.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import PropTypes from "prop-types";
+import styled from "styled-components/macro";
+
+/**
+ * Primary UI component for user interaction
+ */
+const types = {
+  default: {
+    color: "var(--black)",
+    background: "var(--kutsche)",
```
Comment (lmachens) [inline client/src/stories/Badge.js:11]: Try to avoid German in your code (except in the UI).

File: client/src/stories/Badge.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  box-shadow: ${(props) =>
+    props.type === "timer" ? "var(--insetShadow)" : "var(--shadow)"};
+
+  color: ${(props) => types[props.type].color || types.default.color};
+  background-color: ${(props) =>
+    types[props.type].background || types.default.background};
+`;
+
+export const Badge = ({ type, label }) => {
+  console.log({ label, type });
```
Comment (lmachens) [inline client/src/stories/Badge.js:67]:
```suggestion
```

**Review by lmachens — COMMENTED** [review]
I didn't review everything because this pull request is way too big for a good review. 
Please work in smaller steps and review it one by one. The commits are good 👍.

There is a misunderstanding in the usage of Storybook and the React app. Maybe it's a good idea to use this as template and create new pull requests of this.

File: client/src/App.js (hunk 3 above)
Comment (bwnstck) [reply client/src/App.js:6]: so how would you dot it? All our components are based in `./src/stories/`
Should we add an additional components folder?

Comment (lmachens) [reply client/src/App.js:6]: Yes. Stories are use cases for your components and demonstrate how you could use them.

**Pull request closed without merging**